src/api/spec/models/obs_factory/distribution_spec.rb

```diff
@@ -1,17 +1,229 @@
+
+      it { expect(subject.count).to eq(1) }
+      it { expect(subject.first.class).to eq(ObsFactory::Request) }
+      it { expect(subject.first.bs_request).to eq(review_request) }
+    end
+
+    context 'group' do
+      let(:group) { create(:group, title: 'Staff') }
+      let(:other_group) { create(:group, title: 'Other') }
+      let!(:review_request) do
+        create(:review_bs_request_by_group,
+               reviewer: group.title,
+               target_project: project.name,
+               target_package: target_package.name,
+               source_project: source_package.project.name,
+               source_package: source_package.name)
+      end
+      let!(:other_review_request) do
+        create(:review_bs_request_by_group,
+               reviewer: other_group.title,
+               target_project: project.name,
+               target_package: target_package.name,
+               source_project: source_package.project.name,
+               source_package: source_package.name)
+      end
+
+      subject { distribution.requests_with_reviews_for_group(group.title) }
+
+      it { expect(subject.count).to eq(1) }
+      it { expect(subject.first.class).to eq(ObsFactory::Request) }
+      it { expect(subject.first.bs_request).to eq(review_request) }
+    end
+  end
+
+  describe '#standard_project' do
+    it { expect(distribution.standard_project.class).to eq(ObsFactory::ObsProject) }
+    it { expect(distribution.standard_project.exclusive_repository).to eq('standard') }
+  end
+
+  describe '#live_project' do
+    context 'with live project' do
+      let!(:live_project) { create(:project, name: 'openSUSE:Factory:Live') }
+
+      it { expect(distribution.live_project.class).to eq(ObsFactory::ObsProject) }
+      it { expect(distribution.live_project.exclusive_repository).to eq('images') }
+    end
+
+    context 'without live project' do
+      it { expect(distribution.live_project).to be_nil }
+    end
+  end
+
+  describe '#images_project' do
+    it { expect(distribution.images_project.class).to eq(ObsFactory::ObsProject) }
+    it { expect(distribution.images_project.exclusive_repository).to eq('images') }
+  end
+
+  describe '#ring_projects' do
+    context 'with ring projects' do
+      let!(:ring_project) { create(:project, name: 'openSUSE:Factory:Rings') }
+      let!(:ring_project_0) { create(:project, name: 'openSUSE:Factory:Rings:0-Bootstrap') }
+      let!(:ring_project_1) { create(:project, name: 'openSUSE:Factory:Rings:1-MinimalX') }
+      let!(:ring_project_2) { create(:project, name: 'openSUSE:Factory:Rings:2-TestDVD') }
+      let(:ring_project_names) do
+        ['openSUSE:Factory:Rings:0-Bootstrap',
+         'openSUSE:Factory:Rings:1-MinimalX',
+         'openSUSE:Factory:Rings:2-TestDVD']
+      end
+
+      it { expect(distribution.ring_projects.count).to eq(3) }
+      it { expect(distribution.ring_projects.map(&:name)).to eq(ring_project_names) }
```

Comment:
Have you tried using [contain_exactly](https://relishapp.com/rspec/rspec-expectations/v/3-7/docs/built-in-matchers/contain-exactly-matcher)?